### ISSUE TITLE
Connections to prior work lacking + How are these strategies unique to security

### ISSUE THREAD

**Issue by jssmith1** [DeveloperLiberationFront/iTrustInterviews#415]
R1:
>My main complaint is a lack of connection with the other papers that
list questions, especially [27][28][42]. Those papers looked at
different tasks than the current paper, which brings up the issue of
to what extent would each of a programmer's tasks result in separate
lists of questions? What percent of your questions are unique to
security issues? It seems like lots of your questions (like all the
ones about control flow and data flow, e.g., 3.3.4) would be identical
to previously reported. Are there differences in these areas? If so,
why? Are the strategies to solve them the same or different?

>Also, are your questions specifically "hard-to-answer" [28] questions,
or a mix of easy and hard questions?

 >3.3.6 sounds like reachability questions. Is there any difference from
[27]?

**Comment by jssmith1** [comment]
Editor:

>Both R1 and R2 believed that the paper needed a more nuanced discussion of prior work. There is summary in the introduction and discussion, but the paper really needs to go beyond just summary, describing how the discoveries in this paper add to, refine, or reject discoveries in prior work. I think most reviewers viewed this as a critical but minor revision.

>I agree with R3 and believe the analysis, results, and discussion could go deeper in trying to distinguish between the discoveries that are specific to security vulnerabilities and the discoveries that are more general to software development. This is tied to the shallow discussion of prior work, which, if deeper, would likely reveal some of the discoveries for RQ2 and RQ3 to be replications of prior work, and not new. This nuance, however, would likely also be balanced by a deeper understanding of what makes security vulnerabilities different from other kinds of defects.

**Comment by jssmith1** [comment]
R2:

> You state in your Related Work section that there are information needs studies with similar results. It would be helpful if you discuss the differences between the questions developers ask during „normal“ bug fixing or program understanding and „fixing“ security bugs. This discussion may help to answer the question if there are security bugs related questions/strategies/assumptions.

**Comment by jssmith1** [comment]
Do a mapping in the results sub sections to relate our work to previous work more. Where we can't draw a mapping that suggest it might be unique to security.

Look at hard questions to see how to relate to that. Possibly add a col to summary figure to say what's easy or hard.
- Hard to answer questions were identified by the survey "What other hard to answer questions about code can you think of"

Categories might be the same, but unique that they are more specific. 
What questions occur in other studies that don't occur in our context?

Might want to add a summary paragraph saying which of the security needs/strategies are similar, and which are unique. There are some from other papers that don't occur here.

Questions might be the same, but interpretation is different

Add a discussion section to address

**Comment by jssmith1** [comment]
R3:

>4- The strategies the authors finally provide are too general and could, e.g., be used to optimize IDEs rather than security tools.

>- One interesting aspect would be to see how the reported behavior compare to a traditional (i.e., not focused on security) static analyzer.

**Comment by jssmith1** [comment]
Another paper to bring in here:

Asking and Answering Questions during a Programming Change Task on Pharo Language

**Comment by jssmith1** [comment]
Re strategies are too general and could be used to e.g. optimize IDE:
- The fact that similar questions appear in this context compared with more general contexts is unsurprising and, indeed, promising. Tools and techniques that help answer those general questions might benefit developers who are working on some security tasks. Transferability

**Comment by jssmith1** [comment]
Analysis of questions in related works
https://docs.google.com/spreadsheets/d/1_hi-Uc9kzfS__D1KUPS7l50udPfaWI-lBsoetkT99lA/edit#gid=0

**Comment by jssmith1** [comment]
Questions asked in the presence of errors differ from those asked during regular change tasks. Relate to prior discussion of tools that operate in error state and normal conditions

**Comment by jssmith1** [comment]
Points of comparison in Related Work Section:
- Methodological + Domain / task
- Goals of the study -> leads to different categorizations

In categories sections talk about similar questions from other papers

In new discussion section talk about strategies / info needs unique to security

**Comment by jssmith1** [comment]
- [ ] Also cite, "Are vulnerabilities discovered and resolved like other defects?"
https://www.readcube.com/articles/10.1007/s10664-017-9541-1

**Comment by jssmith1** [comment]
- [x] %TODO: Add more substantive difference. Name of problem reflects where the problem originates Attacker vs unexpected failure.
%Crafted to do more damage etc...

**Comment by jssmith1** [comment]
No related work for Understanding Concepts and Confirming Expectations

**Comment by jssmith1** [comment]
New paper not included... also only tangentially related: What Questions do Requirements Engineers Ask?

**Comment by jssmith1** [comment]
**More connections to related work**
Several reviewers suggested we discuss the related work, specifically the other papers that
list questions, more deeply. To that end, we expanded our discussion in two ways. First, we added a paragraph to each category in the results section. In each of those paragraphs, we relate our categories to findings from previous work. Where possible, we distinguish the security implications from the implications pertinent in general programming contexts. Second, we expanded Section ??? to highlight the methodological and domain differences between our study and previous information needs studies.

**Are they hard to answer?**
Latoza and Myers surveyed developers and asked them to self-report which questions they thought were "hard to answer." We did not ask our participants to retroactively asses which of the questions they asked during our study were hard to answer. Therefore, it's hard to classify any of our questions as hard in the same sense they did.

**Is 3.3.6 reachability?**
As we now discuss, the questions in 3.3.6 could be rephrased as reachability questions. 

**Strategies could be used to optimize the IDE**
